content/blog/2024/css-layout.md

```diff
@@ -0,0 +1,286 @@
+
+## Large sites (and core features) are slow to change
+
+Many developers are working in a context
+with much more conservative legacy support policies --
+often waiting for 'full' support of a new feature
+before they even consider using it in production.
+
+In conversations,
+I especially hear concerns
+about browsers like iOS Safari
+that update more slowly,
+sometimes requiring a much heavier OS update.
+Safari on Mac may update in monthly-ish cycles,
+but users with an old phone
+are more often out-of-date.
+
+Big sites are container ships
+that will get stuck in the Suez Canal
+during high winds,
+or decimate the Francis Scott Key bridge
+when there's a power issue.
+These ships are not easy to maneuver.
+Sometimes they are maintaining essential resources
+for a long tail of users on older devices --
+sometimes they are weighed down
+by the sheer scale of their organization.
+Either way,
+it takes some time to change course.
+
+But it's not just the sites
+that are difficult to move.
+Developers can start sprinkling
+`text-wrap: balance` on headings
+with a 5-minute pull request,
+no matter how large the project.
+But Layout Systems are core features,
+and often rely on third-party tooling.
+It's one thing to install new light fixtures,
+it's another thing to replace
+the entire hull of a ship.
+Even with a smaller team,
+a smaller project,
+and a lighter browser-support matrix --
+it doesn't make sense to re-write your code base
+or design system
+every time a new feature comes out.
+
+We started phasing out Susy
+(our light-weight grid system)
+on OddBird projects in 2015.
+By 2017,
+we were
+[recommending others do the same](http://localhost:8080/talks/no-grid-system/),
+and we
+[officially deprecated](https://github.com/oddbird/susy/commit/018b5c638ec0726d172af96828e6f1712682283b)
+the project
+on Jul 14, 2020.
+Four years later,
+[OddBird.net/susy](/susy/)
+still has the highest traffic of anything on our site,
+and the `susy` package gets
+[nearly 19k weekly downloads on npm](https://www.npmjs.com/package/susy).
+
+## Learn CSS grid (yes, you need it)
+
+There's no rush to rip out
+all your media queries,
+and replace them with containers.
+You'll be fine waiting for
+_widely available_ support
+and your next scheduled re-factor.
+
+But if you're still avoiding grid --
+whatever your reasons --
+you are, in fact, _missing out_.
+CSS grid is one of the best features in CSS,
+and one of the biggest time-savers
+on every site we build.
+
+I can imagine
+it's hard to re-write
+a system built on flexbox.
+Flexbox is a great feature,
+and part of the web platform.
+What more do you need?
+
+But flexbox is designed for
+content-out distribution,
+not system-wide page layouts.
+Using flexbox alone
+is like only using `inline` text,
+without paragraphs, divs, and other `block` elements.
+You're using half the system,
+and _it's the more complicated half to manage_.
+
+At some point
+people looked at Grid and thought
```

Comment:
```suggestion
people looked at grid and thought
```
I think grid is lowercase in most instances in the post. Maybe we capitalize on in section headings or when it's CSS Grid?